CMakeLists.txt

```diff
@@ -18,6 +18,7 @@ option(BUILD_DOCUMENTATION "Create HTML based API documentation (requires Doxyge
 option(RUN_LONG_UNIT_TESTS "Tests that takes more than 5 minutes to complete. No effect if BUILD_TESTING is OFF" OFF)
 option(BUILD_STORAGE_SAMPLES "Build sample application for Azure Storage clients" OFF)
 option(BUILD_PERFORMANCE_TESTS "Build the performance test library" OFF)
+option(STATIC_CRT "Set to ON to link SDK with static CRT on Windows (use MSVC's /MT switch)." OFF)
```

Comment:
Should this be something like `MSVC_USE_STATIC_CRT` as this option doesn't apply to other STLs?

Reply:
I am not sure, maybe that would apply to MinGW, which is the GCC compiler on Windows. Maybe Intel compiler has the similar option. I'd keep MSVC out of the name, libcurl does it too.

Reply:
@CaseyCarter Is the static crt something specific to Windows only?  Should we specify this as MSVC or should it be generic for all compilers

Reply:
Certainly there are other platforms which provide static libraries and simple ways to link to them. `g++ -static meow.cpp` on linux, for example, produces a statically linked binary. However, `-static` is not IIRC equivalent to `/MT`: the former refuses to link with any DSOs, whereas the latter specifically links the C and C++ runtimes statically but is happy to link with other DLLs. Note that CMake added support for `/MT` in 3.15 via [the MSVC_RUNTIME_LIBRARY property](https://cmake.org/cmake/help/latest/prop_tgt/MSVC_RUNTIME_LIBRARY.html).

Reply:
Thank you all! Resolved.